src/app/index.ts

```diff
@@ -101,147 +98,89 @@ module.exports = yo.extend({
-      let durationForManifestOnly = (endForManifestOnly - startForManifestOnly) / 1000;
-
       /**
        * Configure user input to have correct values
        */
       this.project = {
-        folder: answerForFolder.folder,
+        folder: null,
         name: this.options.name || answerForName.name,
-        host: this.options.host || answerForHost.host,
-        framework: this.options.framework || null,
-        isManifestOnly: answerForManifestOnly.isManifestOnly
+        host: answerForHost.host || this.options.host,
+        projectType: this.options.projectType || null,
+        isManifestOnly: false
       };
-      if (answerForManifestOnly.isManifestOnly) {
-        this.project.framework = 'manifest-only';
+
+      // Set folder to specified name if 'output' option is passed as an argument
+      if (this.options.output != null) {
+        this.project.folder = this.options.output;
+      }
+      else{
+        this.project.folder = this.project.name;
       }
-      if (this.options.framework != null) {
-        if (this.options.framework === 'manifest-only') {
+  
+      // Set isManifestOnly flag to true if project-type argument is 'manifest-only'
+      if (this.options.projectType === manifestOnly) {
           this.project.isManifestOnly = true;
-        } else {
-          this.project.isManifestOnly = false;
-        }
       }
 
-      /** askForTs and askForFramework will only be triggered if it's not a manifest-only project */
-      /** use TypeScript for the project */
-      let startForTs = (new Date()).getTime();
-      let askForTs = [
-        {
-          name: 'ts',
-          type: 'confirm',
-          message: 'Would you like to use TypeScript?',
-          default: true,
-          when: (this.options.js == null) && (!this.project.isManifestOnly) && (this.options.framework !== 'react')
-        }
-      ];
-      let answerForTs = await this.prompt(askForTs);
-      let endForTs = (new Date()).getTime();
-      let durationForTs = (endForTs - startForTs) / 1000;
-      if (!(this.options.js == null)) {
+      // Set js flag to true if 'js' option is passed as an argument
+      if (this.options.js != null) {
         this.project.ts = !this.options.js;
       }
       else {
-        this.project.ts = answerForTs.ts || false;
+        this.project.ts = true;
       }
-      if (this.options.framework === 'react') {
+      if (this.options.projectType === 'react') {
         this.project.ts = true;
       }
 
-      /** technology used to create the addin (html / angular / etc) */
-      let startForFramework = (new Date()).getTime();
-      let askForFramework = [
+      /** technology used to create the addin (jquery / angular / etc) */
+      let startForProjectType = (new Date()).getTime();
+      let askForProjectType = [
         {
-          name: 'framework',
-          message: 'Choose a framework:',
+          name: 'projectType',
+          message: 'Choose a project-type:',

```

Comment:
nit - Since this is just an English prompt can we change "project-type" to "project type"?

Reply:
fixed locally